### ISSUE TITLE
Regression: OniNaki now crashes on startup with an audout related error

### ISSUE THREAD

**Issue by EmulationFanatic** [Ryujinx/Ryujinx#1966]
### What's the issue you encountered?

This game used to boot and only had some minor graphical issues. Attempting to boot [OniNaki](https://github.com/Ryujinx/Ryujinx-Games-List/issues/784) on current master 1.0.6421 results in this crash (note: this occurs with both OpenAL and SoundIO backends):

```
HLE.OsThread.8 Application : Unhandled exception caught: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.OverflowException: Array dimensions exceeded supported range.
   at Ryujinx.HLE.HOS.Services.Audio.AudioOutManager.IAudioOut.AppendAudioOutBufferImpl(ServiceCtx context, Int64 position) in C:\projects\ryujinx\Ryujinx.HLE\HOS\Services\Audio\AudioOutManager\IAudioOut.cs:line 118
   at Ryujinx.HLE.HOS.Services.Audio.AudioOutManager.IAudioOut.AppendAudioOutBufferAuto(ServiceCtx context) in C:\projects\ryujinx\Ryujinx.HLE\HOS\Services\Audio\AudioOutManager\IAudioOut.cs:line 108
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Ryujinx.HLE.HOS.Services.IpcService.CallMethod(ServiceCtx context) in C:\projects\ryujinx\Ryujinx.HLE\HOS\Services\IpcService.cs:line 114
   at Ryujinx.HLE.HOS.Services.ServerBase.Process(Int32 serverSessionHandle, UInt64 recvListAddr) in C:\projects\ryujinx\Ryujinx.HLE\HOS\Services\ServerBase.cs:line 208
   at Ryujinx.HLE.HOS.Services.ServerBase.ServerLoop() in C:\projects\ryujinx\Ryujinx.HLE\HOS\Services\ServerBase.cs:line 132
   at Ryujinx.HLE.HOS.Kernel.Threading.KThread.ThreadStart() in C:\projects\ryujinx\Ryujinx.HLE\HOS\Kernel\Threading\KThread.cs:line 1031
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

### How can the issue be reproduced?

Boot up the game on current master.

### Log file

[ONINAKI.log](https://github.com/Ryujinx/Ryujinx/files/5873830/ONINAKI.log)

### Environment?

- Ryujinx version: 1.0.6421
- Game version: 1.0.0
- System Specs:
  - OS: Windows 10 20H2
  - CPU: Ryzen 7 5800X
  - GPU: NVIDIA GTX 1080
  - RAM: 32GB
- Applied Mods : No

**Comment by Shahil-Ayato** [comment]
@EmulationFanatic this issue is fixed with the latest ryujinx master version.

**Comment by gdkchan** [comment]
Fixed on #2007
